### ISSUE TITLE
Encountered several problems

### ISSUE THREAD

**Issue by sheriby** [kevinhwang91/rnvimr#12]
- **Ranger-devicons** works well when using ranger in terminal. However, I cannot use it in rnvimr.
- Similarly, I cannot **preview images** well using rnvimr. Pictures are very big. They are not in correct position and even part of them are out of the screen.  

I am using Archlinux + dwm now. How can I fix these problems?

**Comment by kevinhwang91** [comment]
> * **Ranger-devicons** works well when using ranger in terminal. However, I cannot use it in rnvimr.
You should syn your ranger plugins by using `:RnvimrSync`. Make sure your ranger-devicons exists in `$HOME/.config/nvim/plugged/rnvimr/ranger`.

> * Similarly, I cannot **preview images** well using rnvimr. Pictures are very big. They are not in correct position and even part of them are out of the screen.

rnvimr only calibrated preview image for uberzug for now. 
If use uberzug, this may happen when your window size is small. Preview image works best in tmux because of tmux api. 

> I am using Archlinux + dwm now. How can I fix these problems?
I have a friend using Archlinux + dwm, rnvimr works flawless in his env.

**Comment by kevinhwang91** [comment]
Had your read README to update the lastest version of uberzug?

https://github.com/kevinhwang91/rnvimr/commit/84dc67738ca23918cd15d99fbfbf4ca44bb66c12

**Comment by sheriby** [comment]
Thank you. I have solved the first problem.  
Actually, I am using `uberzug` to preview image. What consused me is not rander pictures but pictures not in place...I try to adjust g:rnvimr_layout but no results.
Maybe your friend is theniceboy?? I know this plugin from his nvim repo.

**Comment by sheriby** [comment]
I have installed the latest version of `ueberzug`.

**Comment by kevinhwang91** [comment]
> Thank you. I have solved the first problem.
> Actually, I am using `uberzug` to preview image. What consused me is not rander pictures but pictures not in place...I try to adjust g:rnvimr_layout but no results.

Could you upload an image? I can't get your point clearly.

> Maybe your friend is theniceboy?? I know this plugin from his nvim repo.

Yes.


> I have installed the latest version of `ueberzug`.

Install by `yay -S python-ueberzug-git`.

**Comment by sheriby** [comment]
![scrot](https://user-images.githubusercontent.com/49471532/77546898-794fe000-6ee7-11ea-82df-a6edba9c9421.png)

**Comment by sheriby** [comment]
Just as what you see, the picture is no in place.

**Comment by kevinhwang91** [comment]
What's your terminal? Had you tested with others? I have tested under dwm with alacritty which works flawlessly.

**Comment by sheriby** [comment]
I using simple terminal. However, testing in alacritty, nothing changes.

**Comment by sheriby** [comment]
I am using simple terminal. However, testing in alacritty, nothing changes.

**Comment by sheriby** [comment]
I am using simple terminal now. However, testing in alacritty, nothing changes.

**Comment by sheriby** [comment]
![2020-03-25-230739_2158x1408_scrot](https://user-images.githubusercontent.com/49471532/77551572-7b1ca200-6eed-11ea-81d8-b01f7614a0f6.png)
So is it probably my rnvimr configuration problem??

**Comment by kevinhwang91** [comment]
No, there aren't any configurations in rrnvimr refer to this issue.  

Maybe you should check out the version about ueberzug. I found that no command can output the version of ueberzug. You may diff your ueberzug lib and check it out with https://github.com/seebye/ueberzug/commit/d86eeb303b9468226884425853472287a794d9dd

My lib path is `/usr/lib/python3.8/site-packages/ueberzug`

**Comment by sheriby** [comment]
Thanks a lot! I will follow u suggestion to find out the solution.

**Comment by sheriby** [comment]
I haven't found a perfect solution. But I found a personal temporary solution, adjust varibes **start_x, start_y, width and height** of function `calibrate_ueberzug` in `~/.config/nvim/plugged/rnvimr/lib/hacks.py`.
However, resizing rnvimr will break it...XDXD

**Comment by kevinhwang91** [comment]
```python
    def calibrate_ueberzug(self):
        """
        Ueberzug can't capture the calibration of floating window of neovim, fix it.

        """

        client = self.fm.client

        def wrap_draw(self, path, start_x, start_y, width, height):
            win_info = client.request('nvim_win_get_config', 0)
            if not win_info['relative']:
                return
            start_x += win_info['col']
            start_y += win_info['row']

            self.raw_draw(path, start_x, start_y, width, height)

        try:
            # ueberzug is supported by ranger since [b58954d4258bc204c38f635e5209e6c1e2bce743]
            # https://github.com/ranger/ranger/commit/b58954d4258bc204c38f635e5209e6c1e2bce743
            # pylint: disable=import-outside-toplevel
            from ranger.ext.img_display import UeberzugImageDisplayer
        except ImportError:
            pass
        else:
            UeberzugImageDisplayer.raw_draw = UeberzugImageDisplayer.draw
            UeberzugImageDisplayer.draw = wrap_draw
```
This method only capture the top left coordinate of floating window. 
I'm not sure that `nvim_win_get_config` work proper for you. You can enter neovim command line mode (default keymap is <C-\>) to use `echo nvim_win_get_config(0)` in floating window to check out whether the size of floating window is proper for you. If not, please check out the version of your neovim.

You can also test the behavior of previewing image under raw ranger in floating window by [vim-floaterm](https://github.com/voldikss/vim-floaterm).

 In short, make sure that `echo nvim_win_get_config(0)` and the version of ueberzug are correct.

**Comment by sheriby** [comment]
I just adjust the position and size of picture manually. Anyway it exactly works for me if not resizing rnvimr. Probably there's a hidden environmental problem in my computer...finding out and fixing it may be difficult.lollol
```python
    def calibrate_ueberzug(self):
        """
        Ueberzug can't capture the calibration of floating window of neovim, fix it.

        """

        client = self.fm.client

        def wrap_draw(self, path, start_x, start_y, width, height):
            win_info = client.request('nvim_win_get_config', 0)
            if not win_info['relative']:
                return
            #  start_x += win_info['col']
            #  start_y += win_info['row']
            start_y += win_info['row'] *2/3 + 1
            width = width*3/ 4
            height = height*3 / 4

            self.raw_draw(path, start_x, start_y, width, height)

        try:
            # ueberzug is supported by ranger since [b58954d4258bc204c38f635e5209e6c1e2bce743]
            # https://github.com/ranger/ranger/commit/b58954d4258bc204c38f635e5209e6c1e2bce743
            # pylint: disable=import-outside-toplevel
            from ranger.ext.img_display import UeberzugImageDisplayer
        except ImportError:
            pass
        else:
            UeberzugImageDisplayer.raw_draw = UeberzugImageDisplayer.draw
            UeberzugImageDisplayer.draw = wrap_draw

```

**Comment by sheriby** [comment]
Thanks a lot for your patient help!!! Maybe we should close this issue, after all it is my problem not this plugin.

**Comment by kevinhwang91** [comment]
You can also test under tmux. Old and latest ueberzug will use tmux api to fix X11. Or change your X11 resolution to 4k or 1080p.



> Thanks a lot for your patient help!!! Maybe we should close this issue, after all it is my problem not this plugin.

This is a strange issue, I'm interested in it. Maybe it's a compatible issue.

**Comment by kevinhwang91** [comment]
If you love this plugin, but can't solve this issue by yourself, you can email me with Chinese.
kevin.hwang@live.com

**Comment by sheriby** [comment]
It is not resolution problem.(My resolution is 2160x1440). 
By `xrandr -s 1920x1080` switching to 1080p, it doesn't work. Pictures are still not in place.
And then I switch to Deepin DDE and Deepin terminal, it doesn't work as well.
I reinstall `ueberzug` by `pip install ueberzug --user` again, nothing changes.

**Comment by kevinhwang91** [comment]
> It is not resolution problem.(My resolution is 2160x1440).
> By `xrandr -s 1920x1080` switching to 1080p, it doesn't work. Pictures are still not in place.
> And then I switch to Deepin DDE and Deepin terminal, it doesn't work as well.
> I reinstall `ueberzug` by `pip install ueberzug --user` again, nothing changes.

我几乎肯定是你的ueberzug的版本问题，当我使用`pip install ueberzug --user`就会出现问题，请使用`yay -S python-ueberzug-git`或者git clone master并手动安装。

**Comment by sheriby** [comment]
非常感谢，从AUR中安装`ueberzug`之后，这个问题被解决了。
同时也非常抱歉，我应该早点听从你的建议从AUR中安装的，只是一开始我使用`yay -S python-ueberzug-git`时出现了很多安装问题。因为虽然卸载了`pip uninstall ueberzug`，但是很多依赖库没有被卸载。当我手动卸载所有的依赖库之后再次从AUR中安装的时候，成功了。
再次非常感谢您的帮助，我非常喜欢这个插件。
